doc/reference/modules/performance.xml

```diff
@@ -1216,131 +1216,194 @@ sessionFactory.EvictCollection("Eg.Cat.Kittens");]]></programlisting>
+        <programlisting><![CDATA[var queries = s.CreateQueryBatch()
+    .Add<Item>(
+        s.CreateCriteria(typeof(Item))
+         .Add(Expression.Gt("Id", 50))
+         .SetFirstResult(10))
+    .Add<long>(
+        s.CreateCriteria(typeof(Item))
+         .Add(Expression.Gt("Id", 50))
+         .SetProject(Projections.RowCount()));
+var items = queries.GetResult<Item>(0);
+var count = queries.GetResult<long>(1).Single();]]></programlisting>
 
         <para>
-            The result is a list of query results, ordered according to the order of queries
-            added to the multi query. Named parameters can be set on the multi query, and are
-            shared among all the queries contained in the multi query, like this:
+            You can add <literal>ICriteria</literal> or <literal>DetachedCriteria</literal> to the query batch.
+            In fact, using DetachedCriteria in this fashion has some interesting implications.
         </para>
+        <programlisting><![CDATA[DetachedCriteria customersCriteria = AuthorizationService.GetAssociatedCustomersQuery();
+IQueryBatch queries = session.CreateQueryBatch()
+    .Add<Customer>(customersCriteria)
+    .Add<Policy>(DetachedCriteria.For<Policy>()
+        .Add(Subqueries.PropertyIn("id",
+            CriteriaTransformer.Clone(customersCriteria)
+                .SetProjection(Projections.Id())
+            )));
 
-        <programlisting><![CDATA[IList results = s.CreateMultiQuery()
-    .Add(s.CreateQuery("from Item i where i.Id > :id")
-          .SetFirstResult(10))
-    .Add("select count(*) from Item i where i.Id > :id")
-    .SetInt32("id", 50)
-    .List();
-IList items = (IList)results[0];
-long count = (long)((IList)results[1])[0];]]></programlisting>
+IList<Customer> customers = queries.GetResult<Customer>(0);
+IList<Policy> policies = queries.GetResult<Policy>(1);]]></programlisting>
 
         <para>
-            Positional parameters are not supported on the multi query, only on the individual
-            queries.
+            We get a query that represents the customers we can access, and then we can utilize this
+            query further in order to perform additional logic (getting the policies of the customers we are
+            associated with), all in a single database round-trip.
         </para>
 
         <para>
-            As shown above, if you do not need to configure the query separately, you can simply
-            pass the HQL directly to the <literal>IMultiQuery.Add()</literal> method.
+            The query batch also supports QueryOver and sql queries. All kind of queries can be mixed in the
+            same batch.
         </para>
 
+        <programlisting><![CDATA[using NHibernate.Multi;
+
+...
+
+var queries = s.CreateQueryBatch()
+    .Add("queryOverList", s.QueryOver<Item>().Where(i => i.Category == "Food"))
+    .Add<long>("sqlCount",
+        s.CreateSQLQuery("select count(*) as count from Item i where i.Category = :cat")
```

Comment:
As this "as count" can cause issues on some dialects maybe we shouldn't use it documentation.